corehq/motech/repeaters/repeater_generators.py

```diff
@@ -364,12 +365,32 @@ def _set_constant_properties(self, case, config):
         for name, value in config.constant_properties:
             case.case_json[name] = value
 
-    def _set_referral_properties(self, case, original_case_id):
+    def _set_referral_properties(self, case, original_case_id, original_case_json):
         # make sure new case is open
         case.closed = False
         case.case_json['cchq_referral_source_domain'] = self.repeater.domain
         case.case_json['cchq_referral_source_case_id'] = original_case_id
 
+        domain_history = original_case_json.get('cchq_referral_domain_history', '').split()
+        id_history = original_case_json.get('cchq_referral_case_id_history', '').split()
+
+        if 'cchq_referral_source_domain' in original_case_json and not domain_history:
+            # bootstrap for cases that have already been transferred at least once prior to
+            # addition of the history properties
+            previous_source_domain = original_case_json.get('cchq_referral_source_domain')
+            previous_source_case_id = original_case_json.get('cchq_referral_source_case_id')
+            domain_history.append(f"_unknown_ {previous_source_domain}")
+            id_history.append(f"_unknown_ {previous_source_case_id}")
```

Comment:
It feels a little weird to have space-separated items in this list.  Also, since the lists are empty, this implementation to me more obviously initializes the histories, but feel free to take it or leave it:
```suggestion            
            domain_history = ["_unknown_", previous_source_domain]
            id_history = ["_unknown_", previous_source_case_id]
```

Reply:
since the lists end up being joined with a space at the end the effect is the same. I did it this way first and then thought about the way you've suggested but decided not to. I guess it does make it clearer.